extensions/podman/packages/extension/src/podman-install.ts

```diff
@@ -672,12 +667,17 @@ export class VirtualMachinePlatformCheck extends BaseCheck {
 }
 
 abstract class WindowsCheck extends BaseCheck {
+  #powerShellClient?: Promise<PowerShellClient> = undefined;
+  protected async getPowerShellClient(): Promise<PowerShellClient> {
+    if (!this.#powerShellClient) {
+      this.#powerShellClient = getPowerShellClient();
+    }
+    return this.#powerShellClient;
+  }
+
   async isUserAdmin(): Promise<boolean> {
     try {
-      const { stdout: res } = await extensionApi.process.exec('powershell.exe', [
-        '$null -ne (whoami /groups /fo csv | ConvertFrom-Csv | Where-Object {$_.SID -eq "S-1-5-32-544"})',
-      ]);
-      return res.trim() === 'True';
+      return (await this.getPowerShellClient()).isUserAdmin();
```

Comment:
Try/catch block is not needed around client calls, because client methods already handle the exceptions.

Reply:
getPowerShellClient may return an exception in the future when we start implementing PowerShell 7 support